dds/DCPS/transport/framework/TransportReassembly.cpp

```diff
@@ -270,17 +270,28 @@ TransportReassembly::reassemble_i(const SequenceRange& seqRange,
       data.header_.sequence_.getValue(), OPENDDS_STRING(conv).c_str()));
   }
 
+  // Purge out expired fragments.
+  const MonotonicTimePoint now = MonotonicTimePoint::now();
+  for (ExpirationQueue::iterator pos = expiration_queue_.begin(), limit = expiration_queue_.end();
+       pos != limit && pos->first <= now;) {
+    fragments_.erase(pos->second);
+    expiration_queue_.erase(pos++);
+  }
+
   const FragKey key(data.header_.publication_id_, data.header_.sequence_);
 
   FragInfoMap::iterator iter = fragments_.find(key);
   if (iter == fragments_.end()) {
-    fragments_[key] = FragInfo(firstFrag, FragRangeList(1, FragRange(seqRange, data)), total_frags);
+    const MonotonicTimePoint expiration = MonotonicTimePoint::now() + TimeDuration(300);
+    fragments_[key] = FragInfo(firstFrag, FragRangeList(1, FragRange(seqRange, data)), total_frags, expiration);
+    expiration_queue_.insert(std::make_pair(expiration, key));
     // since this is the first fragment we've seen, it can't possibly be done
     VDBG((LM_DEBUG, "(%P|%t) DBG:   TransportReassembly::reassemble() "
       "stored first frag, returning false (incomplete)\n"));
     return false;
   } else {
     if (iter->second.complete_) {
+      // TODO: Why is this returning false?
```

Comment:
We only return true when there's something to process. If it's been completed already, someone else has already gotten the "true".

Reply:
I updated the comment on my branch